Source/Utils/ParticleUtils.cpp

```diff
@@ -40,10 +40,16 @@ namespace ParticleUtils {
             // Pass lambda function that returns the cell index
             [=] AMREX_GPU_HOST_DEVICE (const ParticleType& p) noexcept -> IntVect
             {
-                return IntVect(AMREX_D_DECL(
-                                   static_cast<int>((p.pos(0)-plo[0])*dxi[0] - lo.x),
-                                   static_cast<int>((p.pos(1)-plo[1])*dxi[1] - lo.y),
-                                   static_cast<int>((p.pos(2)-plo[2])*dxi[2] - lo.z)));
+#if AMREX_SPACEDIM == 3
+                return IntVect(
+                   static_cast<int>((p.pos(0)-plo[0])*dxi[0] - lo.x),
+                   static_cast<int>((p.pos(1)-plo[1])*dxi[1] - lo.y),
+                   static_cast<int>((p.pos(2)-plo[2])*dxi[2] - lo.z));
+#else
+                return IntVect(
+                    static_cast<int>((p.pos(0)-plo[0])*dxi[0] - lo.x),
+                    static_cast<int>((p.pos(1)-plo[1])*dxi[1] - lo.z)); // double-check
```

Comment:
not 100% if I need to access 1 or 2 in plo, dxi and if `lo.z` is right or `lo.y` in that case

Reply:
at the moment:
```
working on test: leveling_thinning
   re-making clean...
   building...
   found pre-built executable for this test
   copying files to run directory...
   running the test...
   mpiexec -n 4 ./main2d.gnu.TEST.TPROF.MTMPI.OMP.OPMD.ex inputs_leveling_thinning  diag1.file_prefix=leveling_thinning_plt   amrex.abort_on_unused_inputs=1
   WARNING: unable to open the job_info file
   doing the analysis...
   WARNING: analysis failed...
./analysis_leveling_thinning.py  leveling_thinning_plt00008
difference between expected and actual final number of particles (1st species): 164.7159763313539
tolerance: 786.425927850232
Traceback (most recent call last):
  File "./analysis_leveling_thinning.py", line 66, in <module>
    assert(np.amax(np.abs(w-final_weight)) < relative_tol*final_weight )
AssertionError
```

Reply:
```suggestion
                    static_cast<int>((p.pos(1)-plo[1])*dxi[1] - lo.y)); // double-check
```

It's actually `lo.y`, since this is an amrex structure that considers `y` to be the second direction.

Reply:
Yes, I agree with @atmyers. Actually, I think the existing code is already correct (but maybe you prefer to have explicit `#ifdef` rather than `AMREX_D_DECL`).

For RZ, we might have to be careful how we use this function because particles can be in the same cell but far apart spatially (if they have a different theta) so we may not want to directly do collisions/merging with them in this case.